llvm/tools/llvm-profgen/ProfileGenerator.cpp

```diff
@@ -1183,11 +1183,9 @@ void ProfileGeneratorBase::extractProbesFromRange(
     do {
       const AddressProbesMap &Address2ProbesMap =
           Binary->getAddress2ProbesMap();
-      auto It = Address2ProbesMap.find(IP.Address);
-      if (It != Address2ProbesMap.end()) {
-        for (const MCDecodedPseudoProbe &Probe : It->second) {
-          ProbeCounter[&Probe] += Count;
-        }
+      for (const MCDecodedPseudoProbe &Probe :
+           Address2ProbesMap.find(IP.Address)) {
```

Comment:
This is a hot path, but now we change the O(1) `unordered_map` search to a binary search. Not sure if it can cause noticeable regression. I will run an end-to-end test on llvm-profgen.